adapters/repos/db/lsmkv/bucket.go

```diff
@@ -418,6 +418,35 @@ func (b *Bucket) Get(key []byte) ([]byte, error) {
 }
 
 func (b *Bucket) get(key []byte) ([]byte, error) {
+	val, found, err := b.getFromMemory(key)
+	if err != nil {
+		return nil, err
+	}
+
+	if found {
+		return val, nil
+	}
+
+	return b.disk.get(key)
+}
+
+// exists checks whether the object is present in the system and is not deleted.
+func (b *Bucket) exists(key []byte) (bool, error) {
+	val, found, err := b.getFromMemory(key)
+	if err != nil {
+		return false, err
+	}
+	if found {
+		return val != nil, nil
```

Comment:
I think it shouldn't be assumed that nil value is non-existent

Reply:
@jeroiraz I guess that assumes that [`value == nil` indicates that the object is deleted](https://github.com/weaviate/weaviate/pull/7301/files#diff-9658bb68e1a9d3f3d4ae83104656f4dc7c6cf22d931592c53971acdda7225e28R448) ⬇️

Reply:
yeah, this was done before: https://github.com/weaviate/weaviate/pull/7301/files#diff-9658bb68e1a9d3f3d4ae83104656f4dc7c6cf22d931592c53971acdda7225e28R462-R465

Reply:
the combination for deleted entries is `found == true && val == nil`

Reply:
yes but that's by concern, I think it should and it's valid to insert a key-value pair with an empty value

Reply:
I can take a look at this issue later but would like to keep this out of the scope of this PR.

This pattern looks like widespread across the whole db, another example: https://github.com/weaviate/weaviate/blob/114b8ffa7865b3582af556c2e4cacd548d4a1d79/adapters/repos/db/lsmkv/segment_group.go#L407

Reply:
hmm,  given this is new code it may be better to stop spreading that assumption, but we could fix the other parts of the code in a different PR for sure